overlord/devicestate/devicestate_gadget_test.go

```diff
@@ -238,11 +240,18 @@ func (s *deviceMgrGadgetSuite) testUpdateGadgetOnCoreSimple(c *C, grade string)
 
 func (s *deviceMgrGadgetSuite) TestUpdateGadgetOnCoreSimple(c *C) {
 	// unset grade
-	s.testUpdateGadgetOnCoreSimple(c, "")
+	encryption := false
+	s.testUpdateGadgetOnCoreSimple(c, "", encryption)
+}
+
+func (s *deviceMgrGadgetSuite) TestUpdateGadgetOnUC20CoreSimpleWithEncryption(c *C) {
+	encryption := true
+	s.testUpdateGadgetOnCoreSimple(c, "dangerous", encryption)
```

Comment:
shouldn't we try to test some behavior difference between the two cases?

Reply:
Added this now. A bit of roundabout approach, but I believe it should be fine.